### PR TITLE
Improved the suggestions for typed holes actions

### LINKED ISSUE #767: Consider showing local bindings for type hole substitutions in 8.4.3
Since GHC does not show local bindings as valid substitutions in 8.4.3, it would be cool to append the `Relevant bindings include` section in the bottom of the list.

 One heuristic could be that if the only valid substitution is `undefined` or the `error*` functions, then appending the local bindings for help may be a good idea.

Additionally, I think it would be helpful to show the type of the binding as part of the substitution. For instance, it could read as follows:

```
Substitute hole (ShellCheck.AST.Token) with allFlags (ShellCheck.AST.Token -> [String])
Substitute hole (ShellCheck.AST.Token) with hasYesOption (ShellCheck.AST.Token -> Bool)
Substitute hole (ShellCheck.AST.Token) with undefined
```

### PR DESCRIPTION
**Pull request by lorenzo**
This implements what is described in #767

I took the liberty of renaming one test file, so I could run the tests
from inside ghci. Without the rename, it would complain about duplicate
module names

closes #767

### CODE REVIEW

File: test/functional/FunctionalCodeActionsSpec.hs [hunk 1]
```diff
@@ -233,6 +233,31 @@ spec = describe "code actions" $ do
             \foo :: [Int] -> Int\n\
             \foo x = maxBound"
 
+      it "shows more suggestions" $ when ghc84 $
```
Comment (alanz) [inline test/functional/FunctionalCodeActionsSpec.hs]: Can we add parsing for any of the other GHC versions too?

Comment (lorenzo) [reply test/functional/FunctionalCodeActionsSpec.hs]: Sure, I can look into that

**Comment by lorenzo** [comment]
@alanz done

File: test/functional/FunctionalCodeActionsSpec.hs [hunk 2]
```diff
@@ -17,6 +17,7 @@ import qualified Language.Haskell.LSP.Types as LSP
 import qualified Language.Haskell.LSP.Types.Capabilities as C
 import Test.Hspec
 import TestUtils
+import Data.Semigroup
```
Comment (Bubba) [inline test/functional/FunctionalCodeActionsSpec.hs]: You can use import `Data.Monoid ((<>))` which works on both 8.4 and 8.2

**Comment by lorenzo** [comment]
@Bubba Fixed

**Pull request merged**